crates/typst-library/src/visualize/todo.rs

```diff
@@ -0,0 +1,97 @@
+use ecow::eco_format;
+
+use crate::diag::{warning, SourceResult};
+use crate::engine::Engine;
+use crate::foundations::{
+    elem, Content, Packed, Show, ShowSet, StyleChain, Styles, Synthesize,
+};
+use crate::introspection::Locatable;
+use crate::layout::Em;
+use crate::text::{FontWeight, TextElem, TextSize};
+
+use super::Color;
+
+/// A reminder to implement a feature, add content, or fix a bug.
+///
+/// By default this element is shown as a big bold red "TODO", and produces
+/// a warning with an optional message. You can optionally restyle the shown
+/// text by using the `show` rule, and you can disable the warning by using the
+/// `warn` argument.
+///
+/// ```example
+/// #show todo: it => text(fill: red, size: 16pt, it.message)
+///
+/// #todo(message: "Complete this example.")
+/// ```
+///
+/// # Example
+/// ```example
+/// // Produces a warning and a big bold red "TODO" in the console.
```

Comment:
The comments every time seem unnecessary: there will be a preview on the right, right (even though warnings will be emitted)?

Reply:
it's precisely because the preview will not show warnings that I added comments, since there is no way AFAIK to show warnings in the documentation

Reply:
Then maybe the comments should only mention the warnings?

Otherwise, I would rephrase it as follows:
> Produces a warning in the console and a big bold red "TODO".